Indicators/IndicatorBase.cs

```diff
@@ -98,6 +98,25 @@ public bool Update(T input)
         }
 
         /// <summary>
+        /// Updates the state of this indicator with the given value and returns true
+        /// if this indicator is ready, false otherwise
+        /// </summary>
+        /// <param name="time">The time associated with the value</param>
+        /// <param name="value">The value to use to update this indicator</param>
+        /// <returns>True if this indicator is ready, false otherwise</returns>
+        public bool Update(DateTime time, decimal value)
+        {
+            if (typeof(T) == typeof(IndicatorDataPoint))
+            {
+                return Update((T)Activator.CreateInstance(typeof(T), new object[] { time, value }));
```

Comment:
if `T` is `IndicatorDataPoint`, why not just use `new IndicatorDataPoint(time, value)` ?

Reply:
I tried that, but `Update` didn't accept, since it is looking for an instance of `T`.